### PR TITLE
tap-appstore-01

### PR DESCRIPTION
**Pull request by kamalsingh01**
## Description
Code changes made to tap-appstore/__init__.py forked from [Tap-Appstore](https://github.com/miroapp/tap-appstore/tree/master) allowing private_key to passed as string, parsing changes in some field types and inclusion of required stream. 
## Additional Notes
All changes can be found under #ncw
## Change Type
:white_large_square: New feature (non-breaking change which adds functionality)
:white_large_square: Bug fix (non-breaking change which fixes an issue)
:white_check_mark: Refactor
:white_large_square: Breaking change (fix or feature that would cause existing functionality to not work as expected)
:white_large_square: This change requires a documentation update (edited)

### CODE REVIEW

**Review by mredaoudi — APPROVED** [review]

File: tap_appstore/__init__.py [hunk 1]
```diff
@@ -221,20 +227,28 @@ def query_report(api: Api, catalog_entry):
             rep = _attempt_download_report(api, report_filters)
 
             # write records
```
Comment (niko-n) [inline tap_appstore/__init__.py:230]: please add a comment why we are doing a try/except. Sometimes the report for today throws an error.

File: tap_appstore/__init__.py [hunk 2]
```diff
@@ -116,19 +120,21 @@ def discover(api: Api):
...
-        if report:
-            # create and add catalog entry
-            catalog_entry = {
-                'stream': schema_name,
-                'tap_stream_id': schema_name,
-                'schema': schema,
-                'key_properties': []
-            }
-            streams.append(catalog_entry)
+        # ncw - to generate catalog entry for only sales_report, if for all, remove "if"
```
Comment (niko-n) [inline tap_appstore/__init__.py]: please replace 'nwc'.

File: requirement.txt [hunk 3]
```diff
@@ -0,0 +1,24 @@
+appstoreconnect @ git+https://github.com/Peliqan-io/appstoreconnectapi@4aa39df94fcea8de2e892688af2823fae18ab140
```
Comment (niko-n) [inline requirement.txt]: this file not needed

File: tap_appstore/__init__.py [hunk 4]
```diff
@@ -116,19 +120,21 @@ def discover(api: Api):
...
+        if schema_name == "sales_report":
```
Comment (niko-n) [inline tap_appstore/__init__.py]: please check if "metadata" needs to be added on line 136 in the catalog_entry

**Pull request merged**